torch/csrc/autograd/python_anomaly_mode.h

```diff
@@ -18,8 +18,11 @@ struct PyAnomalyMetadata : public AnomalyMetadata {
     dict_ = PyDict_New();
   }
   ~PyAnomalyMetadata() override {
-    pybind11::gil_scoped_acquire gil;
-    Py_DECREF(dict_);
+    // If python is already dead, leak the wrapped python objects
+    if (Py_IsInitialized()) {
```

Comment:
Hmm, shouldn't it be `!_Py_IsFinalizing() && Py_IsInitialized()` (because Python runtime is initialized during finalization part, isn't it?)

Reply:
It is not. There is a full analysis of this here: https://github.com/pytorch/pytorch/blob/589072afa1a196270b0298460c1f419e209da6ea/torch/csrc/utils/pybind.h#L178-L185

I wasn't sure if I should add a link to that in every destructor, should I?